pkg/sidecar/k8s_network.go

```diff
@@ -33,18 +36,92 @@ type K8sNetwork struct {
 	cninet          *libcni.CNIConfig
 	subnet          string
 	netnsPath       string
+	initialized     bool
 }
 
 func (n *K8sNetwork) Close() error {
 	n.nl.Delete()
 	return nil
 }
 
+// getExistingIpRange returns an IP range, if it exists and is attached, that contains the
+// given matchingAddr
+func getExistingIpRange(matchingAddr string) *net.IPNet {
+	// remove mask from matching Addr
+	matchingAddr = strings.Split(matchingAddr, "/")[0]
+
+	logging.S().Infof("Checking local ranges for address %s", matchingAddr)
+
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		logging.S().Errorf("interfaceAddrs error: %+v\n", err.Error())
+		return nil
+	}
+
+	if len(addrs) == 0 {
+		logging.S().Warnf("No InterfaceAddrs exist")
+	} else {
+		// check all interface addresses
+		for _, i := range addrs {
+			_, parsedIp, err := net.ParseCIDR(i.String())
+			if err != nil {
+				logging.S().Errorf(err.Error())
+				continue
+			}
+			iNet := &ptypes.IPNet{IPNet: *parsedIp}
+			matchingNetIp := net.ParseIP(matchingAddr)
+			// does the range contain the target IP?
+			if iNet.Contains(matchingNetIp) {
+				return parsedIp
+			}
+		}
+	}
+	return nil
+
+}
+
+// InitializeNetwork initializes the k8s network. This should be once, before any additional configuration takes place
+func (n *K8sNetwork) InitializeNetwork(ctx context.Context) error {
+	// It is possible an existing address is already assigned by the k8s CNI, and the network doesn't track it
+	// In that case, the easiest thing to do is to simply delete the link - the rest of the configuration code will
+	// handle the rest
+	logging.S().Infow("Initializing k8s network")
```

Comment:
What creates the routes that we have to delete on init?

Reply:
The EKS assigns an IP to each instance via the CNI annotations. When the pod is launched, multus and weave will create the requisite IP interfaces and attach them. The route (singular) that we have to delete is created attached by Weave - because Weave handles the data interface.

Reply:
Theoretically, if we got rid of the the `initialized` bool, wouldn't `InitializeNetwork` become a noop because `getExistingIpRange` wouldn't return anything on a second and any subsequent run? Do I understand this correctly?

Reply:
Not quite.
Let's say that the test desires and receives a new IP address in the data range, `10.32.0.5`. In the subsequent call(s) to `ConfigureNetwork`, `getExistingIpRange` would return the same address, because it contains itself.

Reply:
OK! So then, in the next run we want to reuse the IP address that was created for the previous run rather than create a new one, right? Is that just an optimization?